### PR TITLE
VC JSON Schema compliance with latest spec

### PR DESCRIPTION
**Pull request by decentralgabe**
- reworked schema validation logic to handle both credential and non credential types
- Added tests for all MUST values

### CODE REVIEW

File: credential/schema/vcjsonschema_test.go [hunk 1]
```diff
@@ -49,11 +49,351 @@ func TestValidateCredentialAgainstSchema(t *testing.T) {
...
+			invalidURI := "bad"
+			schemaSubject := schema["credentialSubject"].(map[string]any)
+			schemaSubjectSchema := schemaSubject["jsonSchema"].(map[string]any)
+			schemaSubjectSchema["$id"] = invalidURI
+			schemaSubject["jsonSchema"] = schemaSubjectSchema
+			schema["credentialSubject"] = schemaSubject
+
+			err := IsCredentialValidForJSONSchema(cred, schema, JSONSchemaCredentialType)
+			assert.Error(t, err)
+			assert.Contains(t, err.Error(), "not a valid URI")
```
Comment (andresuribe87) [inline credential/schema/vcjsonschema_test.go:281]: Here and elsewhere
```suggestion
			assert.ErrorContains(t, err, "not a valid URI")
```

File: credential/schema/vcjsonschema_test.go [hunk 2]
```diff
@@ -49,11 +49,351 @@ func TestValidateCredentialAgainstSchema(t *testing.T) {
...
+				"properties": map[string]any{
+					"emailAddress": map[string]any{
+						"type": "string",
+					},
+				},
+				"required": []string{"emailAddress"},
+			},
+		},
+		"required": []string{"credentialSubject"},
+	}
```
Comment (andresuribe87) [inline credential/schema/vcjsonschema_test.go:344]: I would suggest putting the `json` and unmarshalling, instead of instantiating a map.

**Review by andresuribe87 — APPROVED** [review]
Couple nits.

File: credential/schema/vcjsonschema_test.go (hunk 2 above)
Comment (decentralgabe) [reply credential/schema/vcjsonschema_test.go:343]: going to stick with a map since it's easier to maintain IMO

**Pull request merged**